tests/e2e.sh

```diff
@@ -226,3 +320,13 @@ EOF
 kpt fn run .
 grep -qR 'color: orange' .
 grep -qR 'city: toronto' .
+
+# TODO: Add kpt_helm_template_imperative_short and kpt_helm_template_declarative tests after fixing <https://github.com/GoogleContainerTools/kpt/issues/587>
```

Comment:
My suggest would be to have only have e2e tests using kpt and fix this issue to avoid using docker directly. We have sufficient coverage for node-based and docker-base e2e flow above. New tests can be concise (few lines of code) using the kpt CLI. Concern is having 200 lines of e2e in bash for each function is not sustainable/maintainable.

Reply:
I'll keep this as a TODO for now and fix after we make the changes in kustomize and kpt.